File: v2/carga.py
class Carga:
    def __init__(self, carga, volume,sequencia= 0,grupo=None, prioridade_carregamento = False, tipo_box=None,escalada=False):
        self.carga = carga  # Peso da carga
        self.volume = volume  # Volume da carga
        self.grupo = grupo  # Volume da carga  
        self.sequencia = sequencia  # Sequencia carregamento
        self.prioridade_carregamento = prioridade_carregamento
        self.tipo_box = tipo_box
        self.escalada = escalada
        
    def __repr__(self):
        return f"Carga(carga={self.carga}, escalada={self.escalada},sequencia={self.sequencia},volume={self.volume}, grupro={self.grupo}, prioridade_carregamento={self.prioridade_carregamento}, tipo_box={self.tipo_box})\n"
       
    def to_dict(self):
        return {
            "carga": self.carga,
            "volume": self.volume,
            "sequencia": self.sequencia,
            "grupo": self.grupo,
        }

# Função para separar as cargas
def separar_cargas(cargas):
    filhos = [carga for carga in cargas if carga.tipo_box == "filho"]
    grupo = [carga for carga in cargas if carga.tipo_box == "grupo"]
    normal = [carga for carga in cargas if carga.tipo_box == "normal"]
    
    # Agrupar as cargas do tipo "filho" em pares
    filhos_pares = [filhos[i:i + 2] for i in range(0, len(filhos), 2)]
    
    return filhos_pares, grupo, normal

# Função para encontrar a carga com menor sequência e lidar com pares ou carga normal
def carga_com_menor_sequencia(filhos_pares, normal):
    # Combina todas as cargas dos filhos (pares) e normal em uma lista
    todas_as_cargas = [carga for par in filhos_pares for carga in par] + normal
    
    # Encontra a carga com a menor sequência
    carga_menor_sequencia = min(todas_as_cargas, key=lambda carga: carga.sequencia)
    
    # Verifica se a carga com menor sequência é de um par
    for par in filhos_pares:
        if carga_menor_sequencia in par:
            return par  # Retorna o par se a carga for do tipo "filho"
    
    # Se não for do tipo "filho", retorna apenas a carga normal
    return carga_menor_sequencia

File: v2/js/Carga.js
// Carga.js
class Carga {
    constructor(carga, previsaoContainer, sequenciaCarregamento = 0, viagem = null, prioridadeCarregamento = false, tipoBox = null, escalada = false, alocada = false) {
        this.carga = carga;  // Peso da carga
        this.previsaoContainer = previsaoContainer;  // Volume da carga
        this.viagem = viagem;  // Viagem da carga
        this.sequenciaCarregamento = sequenciaCarregamento;  // Sequência do carregamento
        this.prioridadeCarregamento = prioridadeCarregamento;  // Se tem prioridade no carregamento
        this.tipoBox = tipoBox;  // Tipo do box (será preenchido na alocação)
        this.escalada = escalada;  // Se é escalada
        this.alocada = alocada;  // Se é escalada
    }
}

export function separarCargas(cargas) {
    // Separando as cargas pelos tipos
    const filhos = cargas.filter(carga => carga.tipoBox === "filho");
    const grupo = cargas.filter(carga => carga.tipoBox === "grupo");
    const normal = cargas.filter(carga => carga.tipoBox === "normal");

    // Agrupando as cargas do tipo "filho" em pares
    const filhosPares = [];
    for (let i = 0; i < filhos.length; i += 2) {
        filhosPares.push(filhos.slice(i, i + 2));
    }

    return { filhosPares, grupo, normal };
}


export function cargaComMenorSequencia(filhosPares, normal) {
    // Combina todas as cargas dos filhos (pares) e normal em uma lista
    const todasAsCargas = filhosPares.flat().concat(normal);

    // Verifica se há cargas para processar
    if (todasAsCargas.length === 0) {
        return []; // Retorna uma lista vazia se não houver cargas
    }

    // Encontra a carga com a menor sequência
    let cargaMenorSequencia = todasAsCargas.reduce((menor, cargaAtual) =>
        cargaAtual.sequenciaCarregamento < menor.sequenciaCarregamento ? cargaAtual : menor
    );

    // Verifica se a carga com menor sequência pertence a um par
    for (const par of filhosPares) {
        if (par.includes(cargaMenorSequencia)) {
            return par; // Retorna o par como está
        }
    }

    // Se não pertencer a um par, retorna a carga individual dentro de uma lista
    return [cargaMenorSequencia];
}



// Exportando a classe Carga
export { Carga };
